### PR TITLE
Job to execute sending sms/email (Green In Arrears)

### PR DESCRIPTION
**Pull request by cob16**
- Add a new SendAutomatedMessageToTenancy use case (attempts be generic for all automated messages)
- Add Phonelib for number validation
- Refactor config under the x object as per [documentation suggestion](https://guides.rubyonrails.org/v4.2/configuring.html#custom-configuration)
- Update lockfile

### CODE REVIEW

File: app/jobs/application_job.rb [hunk 1]
```diff
@@ -4,7 +4,7 @@ def income_use_case_factory
   end
 
   def run_tenancy_sync_jobs?
-    Rails.application.config.run_tenancy_sync_jobs
+    Rails.configuration.x.run_tenancy_sync_jobs
```
Comment (mrosel) [inline app/jobs/application_job.rb]: this looks like a mis-commit

Comment (cob16) [reply app/jobs/application_job.rb]: > Refactor config under the x object as per [documentation suggestion](https://guides.rubyonrails.org/v4.2/configuring.html#custom-configuration)

File: spec/hackney/income/jobs/send_green_in_arrears_msg_job_spec.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  let(:tenancy_ref) { Faker::Internet.slug }
+  let(:balance) { Faker::Number.decimal(2) }
+
+  before do
+    stub_const('Hackney::Income::GovNotifyGateway', Hackney::Income::StubGovNotifyGateway)
+    stub_const('Hackney::Income::SendAutomatedMessageToTenancy', mock_automated_message)
+    allow(mock_automated_message).to receive(:new).and_return(mock_automated_message)
+  end
+
+  it 'should run the SyncCases use case' do
```
Comment (mrosel) [inline spec/hackney/income/jobs/send_green_in_arrears_msg_job_spec.rb]: should update the name here from `SyncCases` to `SendAutomatedMessage` or similar.

Comment (cob16) [reply spec/hackney/income/jobs/send_green_in_arrears_msg_job_spec.rb]: nice catch

**Pull request merged**